lighthouse-core/gather/driver.js

```diff
@@ -428,6 +428,23 @@ class Driver {
   }
 
   /**
+   * @param {string} name The name of API whose permission you wish to query
+   * @return {!Promise<string>} The state of permissions, resolved in a promise.
+   *    See https://developer.mozilla.org/en-US/docs/Web/API/Permissions/query.
+   */
+  queryPermissionState(name) {
+    const expressionToEval = `
+      (function () {
```

Comment:
any reason to wrap? `navigator.permissions.query({name: ${name}}).then(result => result.state)` is also an expression